sources/SiteCombiner.class.php

```diff
@@ -188,8 +198,18 @@ public function site_js_combine($files, $do_defered)
 	public function site_css_combine($files)
 	{
 		// No files or missing dir then we are done
-		if (empty($files) || !file_exists($this->_archive_dir))
+		if (empty($files) || !file_exists($this->_archive_dir) || !is_writable($this->_archive_dir))
```

Comment:
Looks like a simple new private _checkSpares($files) method would cut out the duplication 😇